### PR TITLE
fix(sdk): restore mtmd_input_text.text_len in VLM tokenize

### LINKED ISSUE #1269: GGML_ASSERT(logits != nullptr)
**Describe the bug**
When trying to infer the model `DavidAU/Qwen3.6-27B-Fable-Fusion-711-Uncensored-Heretic-NM-DAU-NEO-MAX-MTP-GGUF`, it crashes

**To Reproduce**

```
PS C:\Users\maste> geniex infer DavidAU/Qwen3.6-27B-Fable-Fusion-711-Uncensored-Heretic-NM-DAU-NEO-MAX-MTP-GGUF
Model is not currently cached, downloading...
downloading 100% |█████████████████████████████████████████████████████████████████████| (19/19 GB, 41 MB/s)

   Detected model type: vlm
✔  Download success
> TEST
C:/a/GenieX/GenieX/third-party/llama.cpp/common/sampling.cpp:154: GGML_ASSERT(logits != nullptr) failed
```

**Expected behavior**

WORKS

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 11 ARM

**Additional context**

### PR DESCRIPTION
**Pull request by mansiverma897993**
## Summary

2df324a8 accidentally dropped the `text.text_len` assignment that 0aa90fd3 added, leaving the field uninitialized. `mtmd_tokenize` reads `text_len` instead of `strlen(text)`, so every text-only VLM generate crashes: garbage `0` tokenizes an empty string → zero chunks → eval succeeds without decoding → first sample hits `GGML_ASSERT(logits != nullptr)` (sampling.cpp:154). This restores the assignment.

## Test plan

- [x] CPU repro (Qwen3.5-0.8B + mmproj, pinned submodule): `text_len=0` reproduces the exact #1269 assert; with it set, prefill and sampling work

Closes #1269

### CODE REVIEW

**Comment by mengshengwu** [comment]
## Test results

**Summary:** Local repro on Snapdragon X Elite (Windows ARM64). Without the fix, VLM \`generate\` fails immediately with an mtmd-tokenize error; with the fix, cpu / gpu / npu all produce a coherent description of the same image.

Setup: SDK bridge rebuilt against \`sdk/plugins/llama_cpp/src/vlm.cpp\` with and without the one-line \`text.text_len\` restoration, CLI on \`PATH=\$PWD/sdk/pkg-geniex/lib:\$PATH\` so the loader picks up the rebuilt bridge (not the installed release copy under \`%LOCALAPPDATA%/GenieX CLI\`). Prompt: \`./tests/assets/quality_dog.jpg Describe this image.\`. Model: \`google/gemma-4-E4B-it-qat-q4_0-gguf\`.

| Case | \`-c cpu\` | \`-c gpu\` | \`-c npu\` |
|---|---|---|---|
| without fix (\`text_len\` uninitialized) | \`mtmd_tokenize: error: number of media markers in text (0) does not match number of bitmaps (1)\` → \`SDKError(Multimodal generation failed)\` | same error | same error |
| with fix | \`A golden retriever, a medium-to-large, medium-haired dog, is standing on a patch of dirt and fallen leaves outdoors. The dog…\` | \`A full-body photo of a golden retriever dog sitting on a bed of dry leaves outdoors.\` | \`A golden retriever, a medium-sized, sturdy dog with a short, golden-colored coat, is standing in an outdoor setting. The dog is…\` |

Media-marker detection is what the reporter was hitting on \`--compute npu\` — with the uninitialized \`text_len\`, \`mtmd_tokenize\` consumes fewer bytes than the prompt actually contains, the \`<__media__>\` placeholder produced by \`apply_chat_template\` is truncated out, and the image chunk is silently dropped; on this host the fallout is a hard tokenize error, while on the reporter's setup the leftover stack value lands somewhere else in the prompt and the model answers from text alone ("I do not have enough information…"). Same root cause, different presentation of the same corruption.

Note: on this host + gemma-4 the CLIP-on-OpenCL failure mode described in #962 did not reappear — \`-c gpu\` with the current main-line \`mparams.use_gpu = config.n_gpu_layers != 0\` yields a correct caption, so no further change is warranted here.

<details><summary>raw npu run (with fix)</summary>

\`\`\`
[ML] [src/registry.cpp:35] loading plugin from ...\sdk\pkg-geniex\lib\llama_cpp\geniex_plugin.dll
[ML] [plugins/llama_cpp/src/vlm.cpp:185] successfully generated prompt with length=64
[ML] [src/vlm.cpp:121] ErrorCode[0](Success): VlmGenerateOutput(full_text: A golden retriever, a medium-sized, sturdy dog with a short, golden-colored coat, is standing in an outdoor setting. The dog is, profile: prompt_tokens=92 prefill=24.3 tok/s decode=14.0 tok/s)
\`\`\`
</details>

<details><summary>raw npu run (without fix, same command)</summary>

\`\`\`
[ML] [plugins/llama_cpp/src/vlm.cpp:272] using multimodal (mtmd) path with ctx_vision
[ML] [plugins/llama_cpp/src/plugin.cpp:51] mtmd_tokenize: error: number of media markers in text (0) does not match number of bitmaps (1)
[ML] [plugins/llama_cpp/src/vlm.cpp:292] mtmd_tokenize failed
Error: SDKError(Multimodal generation failed)
\`\`\`
</details>

File: sdk/plugins/llama_cpp/src/vlm.cpp [hunk 1]
```diff
@@ -276,6 +276,10 @@ int32_t LlamaVlm::generate(const geniex_VlmGenerateInput* input, geniex_VlmGener
             // prompt_utf8 already has chat template and media markers applied
             mtmd_input_text text;
             text.text          = new_text_portion.c_str();
+            // mtmd_tokenize reads text_len instead of strlen(text); leaving it
+            // unset tokenizes garbage — 0 yields zero chunks, so eval "succeeds"
```
Comment (mengshengwu) [inline sdk/plugins/llama_cpp/src/vlm.cpp]: Rm these commets gen by AI

**Review by mengshengwu — CHANGES_REQUESTED** [review]

**Comment by mengshengwu** [comment]
Cross-linking: I had opened #1272 (now closed) for the same one-line \`text.text_len\` restoration. Since your PR was first with a clean sign-off, closing mine and pushing a small style follow-up onto this branch instead.

Follow-up commit (4c413d82) drops the three-line inline comment: clang-format-18 in \`_lint.yml\` rejects the file with the comment present because it breaks the column-alignment run over \`text.text\` / \`text.text_len\` / \`text.add_special\` / \`text.parse_special\`. Removing the comment lets that block reform. The root-cause explanation already lives in your commit message.

Local repro on Snapdragon X Elite (Windows ARM64), rebuilt SDK bridge on this exact branch, \`google/gemma-4-E4B-it-qat-q4_0-gguf\` VLM with the bundled \`tests/assets/quality_dog.jpg\`:

| Case | \`-c cpu\` | \`-c gpu\` | \`-c npu\` |
|---|---|---|---|
| pre-fix (\`main\`) | \`mtmd_tokenize: number of media markers in text (0) does not match number of bitmaps (1)\` → \`Multimodal generation failed\` | same | same |
| this branch | correct golden-retriever description | correct | correct |

Same uninitialized-read root cause reaches the reporter of #1336 as "I do not have enough information…" on \`--compute npu\` (image chunk silently dropped when the leftover stack value truncates the \`<__media__>\` marker) rather than the assert you reproduced against #1269. One fix covers both.

**Comment by mansiverma897993** [comment]
Done

**Review by mengshengwu — APPROVED** [review]
